### ISSUE TITLE
Context created in ready event not active

### ISSUE THREAD

**Issue by AndreasArvidsson** [talonvoice/talon#630]
If you create a context in a "ready" event that context will not be active directly after Talon has started. If you touch any file in the Talon user directory the context will start working, but directly after startup the `user.aaa` list will be empty. This was easy to confirm with the Talon debug window.

```python
from talon import Module, Context, app

mod = Module()

mod.list("aaa")


def on_ready():
    ctx = Context()
    ctx.lists["user.aaa"] = {"a"}


app.register("ready", on_ready)
```

**Comment by lunixbochs** [comment]
You're letting the context fall out of scope, that's undefined behavior as it may be GC'd. Create it at the top level and set the list in the ready callback.

**Comment by AndreasArvidsson** [comment]
I don't think this is a GC issue. If I add `global ctx` I get the same behavior. It will also continue to work once a file is touched. 

This is of course not a pattern I would normally use. The use case was that we needed an unknown number of context based on csv files on disk for Cursorless. We solved it by just creating a single context and reusing that to populate multiple lists since we didn't need different matches. I just wanted to raise the issue in case someone else tries to do the same thing. 

I just realize that something similar is already used in community. Here the contexts are created in a loop, but not in a ready event which I think is the main cause of the problem I ran into.
https://github.com/talonhub/community/blob/468fb16392e6a9907cb98e2526c1e5cbf3b5fc8d/core/modes/language_modes.py#L80

**Comment by lunixbochs** [comment]
does it work if you create the context outside the callback or not?

**Comment by AndreasArvidsson** [comment]
Yes that works. It's basically what we do in community for the csv files.

**Comment by lunixbochs** [comment]
You should really be creating the context at the top level of the python file where possible, even if you need to populate some attributes later.
